### PR TITLE
feat(snowflake): more ops

### PR DESCRIPTION
**Pull request by cpcloud**
This PR adds some more ops and fixes an uncovered bug in the PySpark backend that showed up after a test refactor.

### CODE REVIEW

File: ibis/backends/snowflake/registry.py [hunk 1]
```diff
@@ -115,12 +114,7 @@ def _extract_url_query(t, op):
         ops.Where: fixed_arity(sa.func.iff, 3),
         # numbers
         ops.RandomScalar: fixed_arity(
-            lambda: sa.func.uniform(
-                sa.cast(0, sa.dialects.postgresql.FLOAT()),
-                sa.cast(1, sa.dialects.postgresql.FLOAT()),
-                sa.func.random(),
-            ),
-            0,
+            lambda: sa.func.uniform(0.0, 1.0, sa.func.random()), 0
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/snowflake/registry.py:117]: Ah ok then I can revert this

[the remark this answers is not in the document]
Comment (krzysztof-kwitt) [reply ibis/backends/snowflake/registry.py:117]: > If min or max, or both, are floating point numbers, the generated numbers are floating point. If both min and max are integers, the generated numbers are integers as well.

https://docs.snowflake.com/en/sql-reference/functions/uniform.html

File: ibis/backends/snowflake/registry.py [hunk 2]
```diff
@@ -159,7 +153,19 @@ def _extract_url_query(t, op):
...
+        ops.NotAll: reduction(lambda arg: ~sa.func.booland_agg(arg)),
+        ops.Any: reduction(sa.func.boolor_agg),
+        ops.NotAny: reduction(lambda arg: ~sa.func.boolor_agg(arg)),
+        ops.BitAnd: reduction(sa.func.bitand_agg),
+        ops.BitOr: reduction(sa.func.bitor_agg),
+        ops.BitXor: reduction(sa.func.bitxor_agg),
+        ops.DateFromYMD: fixed_arity(sa.func.date_from_parts, 3),
+        ops.StringToTimestamp: fixed_arity(sa.func.to_timestamp_tz, 2),
+        ops.RegexExtract: fixed_arity(sa.func.regexp_substr, 3),
+        ops.RegexSearch: fixed_arity(sa.func.regexp_like, 2),
```
Comment (krzysztof-kwitt) [inline ibis/backends/snowflake/registry.py:167]: Alternatively, we can use the REGEX operator, but I don't care.
```suggestion
        ops.RegexSearch: fixed_arity(lambda left, right: left.op('REGEXP')(right), 2),
```

File: ibis/backends/snowflake/registry.py [hunk 3]
```diff
@@ -181,28 +187,16 @@ def _extract_url_query(t, op):
...
     ops.FindInSet,
-    ops.RegexExtract,
-    ops.RegexReplace,
-    ops.RegexSearch,
     ops.StringSplit,
     # ibis.expr.operations.structs
     ops.StructField,
     # ibis.expr.operations.temporal
-    ops.DateFromYMD,
     ops.ExtractMillisecond,
```
Comment (krzysztof-kwitt) [inline ibis/backends/snowflake/registry.py:198]: To implement this function, you have to copy the code from Postgres and change milliseconds to `epoch_mliseconds`.

Comment (krzysztof-kwitt) [reply ibis/backends/snowflake/registry.py:198]: https://docs.snowflake.com/en/sql-reference/functions-date-time.html#supported-date-and-time-parts

File: ibis/backends/snowflake/registry.py [hunk 4]
```diff
@@ -181,28 +187,16 @@ def _extract_url_query(t, op):
...
     ops.IntervalFromInteger,
-    ops.StringToTimestamp,
     ops.TimestampDiff,
     ops.TimestampFromUNIX,
     ops.TimestampFromYMDHMS,
```
Comment (krzysztof-kwitt) [inline ibis/backends/snowflake/registry.py:202]: https://docs.snowflake.com/en/sql-reference/functions/timestamp_from_parts.html

File: ibis/backends/snowflake/registry.py (hunk 3 above)
Comment (cpcloud) [reply ibis/backends/snowflake/registry.py]: Done, thanks!

File: ibis/backends/snowflake/registry.py (hunk 4 above)
Comment (cpcloud) [reply ibis/backends/snowflake/registry.py]: Done!

File: ibis/backends/snowflake/registry.py [hunk 5]
```diff
@@ -181,28 +187,16 @@ def _extract_url_query(t, op):
...
     ops.StringSplit,
     # ibis.expr.operations.structs
     ops.StructField,
     # ibis.expr.operations.temporal
-    ops.DateFromYMD,
     ops.ExtractMillisecond,
     ops.IntervalFromInteger,
-    ops.StringToTimestamp,
     ops.TimestampDiff,
     ops.TimestampFromUNIX,
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/snowflake/registry.py]: Thank you!

**Review by krzysztof-kwitt — APPROVED** [review]

**Pull request merged**